lib/pascal.rb

```diff
@@ -1,36 +1,36 @@
+class PascalTriangle
+  attr_reader :first_row, :second_row, :space
+  def initialize
+    @first_row = [1]
+    @second_row = [1,1]
+    @space = " "
+  end
+  def draw(height)
+    num_spaces = height
+    for row in (1..height)
+      print_row(num_spaces, row)
+      num_spaces -= 1
+    end
+  end
 
-class Pascal
-
-
-  def nth_row(row_num=1)
-    row = [1,1]
-    new_row= []
+  private
+  def print_row(num_spaces, row)
+    puts "#{@space*num_spaces}#{nth_row(row).join(" ")}"
+  end
 
-    if row_num == 1
-      return "1"
-    elsif row_num == 2
-      return row.join(" ")
-    else
-      (row_num-2).times do
-        new_row = []
-        (row.length-1).times do |i|
-            new_row << (row[i]+row[i+1])
-        end
-        row = [1,1]
-        new_row.each do |num|
-          row.insert(1,num)
-        end
-      end
-      return row.join(" ")
-    end
+  def nth_row(row_num)
+    return @first_row  if row_num == 1
+    return @second_row if row_num == 2
+    row = @second_row
+    (row_num - 2).times { row = next_row(row) }
+    row
   end
 
-  def draw_pascal_triangle(height=1)
-    spaces = " "
-    num_spaces = height
-    for row in(1..height)
-      puts "#{spaces*num_spaces}#{nth_row(row)}"
-      num_spaces -=1
+  def next_row(row)
+    new_row = [1,1]
+    (row.length - 1).times do |i|
+      new_row.insert(1,(row[i] + row[i+1]))
     end
+    new_row
```

Comment:
Nicer 👍 . Also, one more candidate for an one-liner `times` block if you prefer it. Make sure you add the white space around various operators `+`, `-`, `*` ... e.g. `row[i + 1]`, between array items e.g. `[1, 1]` and between arguments when you call the methods.